subprojects/core/src/test/groovy/org/gradle/api/internal/project/taskfactory/AnnotationProcessingTasks.java

```diff
@@ -351,13 +367,30 @@ public File getOutputDir() {
         }
     }
 
+    public static class TaskWithNullableOutputDir extends TaskWithAction {
+        @OutputDirectory
+        @Nullable
+        public File getOutputDir() {
+            return null;
+        }
+    }
+
     public static class TaskWithOptionalOutputDirs extends TaskWithAction {
         @SuppressWarnings("deprecation")
         @OutputDirectories
         @org.gradle.api.tasks.Optional
         public File getOutputDirs() {
             return null;
         }
+
+    }
+
+    public static class TaskWithNullableOutputDirs extends TaskWithAction {
+        @OutputDirectories
+        @Nullable
```

Comment:
Nit: would be nicer if we moved `@Nullable` before the `@noun`.

Reply:
Done.